src/unstructured_client/_hooks/custom/clean_server_url_hook.py

```diff
@@ -7,31 +7,61 @@
 from unstructured_client.httpclient import HttpClient
 
 
-class CleanServerUrlSDKInitHook(SDKInitHook):
-    """Hook fixing common mistakes by users in defining `server_url` in the unstructured-client"""
+def clean_server_url(base_url: str) -> str:
+    """Fix url scheme and remove the '/general/v0/general' path."""
+
+    if not base_url:
+        return ""
+    # -- add a url scheme if not present (urllib.parse does not work reliably without it)
+    if "http" not in base_url:
+        base_url = "http://" + base_url
+
+    parsed_url: ParseResult = urlparse(base_url)
+
+    if "api.unstructuredapp.io" in base_url:
+        if parsed_url.scheme != "https":
+            parsed_url = parsed_url._replace(scheme="https")
+
+    # -- path should always be empty
+    return urlunparse(parsed_url._replace(path=""))
 
-    def clean_server_url(self, base_url: str) -> str:
-        """Fix url scheme and remove the '/general/v0/general' path."""
 
-        if not base_url:
-            return ""
-        # -- add a url scheme if not present (urllib.parse does not work reliably without it)
-        if "http" not in base_url:
-            base_url = "http://" + base_url
+def choose_server_url(endpoint_url, client_url, default_endpoint_url) -> str:
```

Comment:
I feel like this entire helper function can be written as a oneliner:
`url = endpoint_url if endpoint_url is not None else (client_url or default_endpoint_url)`
the logic and reasoning behind this can be explained in the docstring.